src/query/executor/transform/exec_test.go

```diff
@@ -0,0 +1,143 @@
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package transform
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/m3db/m3/src/query/block"
+	"github.com/m3db/m3/src/query/functions/utils"
+	"github.com/m3db/m3/src/query/models"
+	"github.com/m3db/m3/src/query/parser"
+	"github.com/m3db/m3/src/query/test"
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/mocktracer"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type testSimpleNode struct {
+}
+
+func (testSimpleNode) Params() parser.Params {
+	panic("implement me")
+}
+
+func (testSimpleNode) ProcessBlock(queryCtx *models.QueryContext, ID parser.NodeID, b block.Block) (block.Block, error) {
+	panic("implement me")
+}
+
+func TestProcessSimpleBlock(t *testing.T) {
+
+	type testContext struct {
+		MockCtrl    *gomock.Controller
+		Controller  *Controller
+		ChildNode   *MockOpNode
+		Node        *MocksimpleOpNode
+		ResultBlock *block.MockBlock
+		SourceBlock block.Block
+		QueryCtx    *models.QueryContext
+	}
+
+	setup := func(t *testing.T) (*testContext, func()) {
+		id := parser.NodeID("foo")
+
+		ctrl := gomock.NewController(t)
+
+		bl := block.NewMockBlock(ctrl)
+
+		controller := &Controller{
+			ID: id,
+		}
+
+		child := NewMockOpNode(ctrl)
+
+		controller.AddTransform(child)
+
+		node := NewMocksimpleOpNode(ctrl)
+
+		return &testContext{
+			MockCtrl:   ctrl,
+			Controller: controller,
+			SourceBlock: test.NewBlockFromValues(
+				models.Bounds{}, [][]float64{{1.0}}),
+			ResultBlock: bl,
+			Node:        node,
+			ChildNode:   child,
+			QueryCtx:    models.NoopQueryContext(),
+		}, ctrl.Finish
+	}
+
+	doCall := func(tctx *testContext) error {
+		return ProcessSimpleBlock(tctx.Node, tctx.Controller, tctx.QueryCtx, tctx.Controller.ID, tctx.SourceBlock)
+	}
+
+	configureSuccessfulNode := func(tctx *testContext) {
+		tctx.Node.EXPECT().Params().Return(utils.StaticParams("foo"))
+		tctx.Node.EXPECT().ProcessBlock(gomock.Any(), gomock.Any(), gomock.Any()).Return(tctx.ResultBlock, nil)
+		tctx.ChildNode.EXPECT().Process(gomock.Any(), gomock.Any(), gomock.Any())
+		tctx.ResultBlock.EXPECT().Close()
+	}
+
+	t.Run("closes next block", func(t *testing.T) {
+		tctx, closer := setup(t)
+		defer closer()
+
+		configureSuccessfulNode(tctx)
+
+		require.NoError(t, doCall(tctx))
+	})
+
+	t.Run("errors on process error", func(t *testing.T) {
```

Comment:
nit: same comment about the `t.Run`, seems to swallow up useful information for failing tests

Reply:
Now that I think about it, mostly had problems with it in tabled tests, when iterating over them... when one of twenty fails it's sometimes a pain to find exactly which test failed exactly where.

Reply:
Ah yes, totally agreed on that! I always try to use searchable names in those cases, which helps, but it can still be annoying.